### PR TITLE
feat: add eval context quality script

### PR DESCRIPTION
**Pull request by wilsvd**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests

[Trello Link](https://trello.com/c/XXXX)

## Screenshots
N/A

### CODE REVIEW

File: scripts/eval_context_quality.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    db = rag.Database()
+
+    collection = db.create_collection()
+
+    # Store documents in collection
+    pdfs_bytes: list[bytes] = []
+    for doc_path in args.doc_paths:
+        with open(doc_path, "rb") as f:
+            pdf_bytes = f.read()
+        pdfs_bytes.append(pdf_bytes)
```
[the remark this answers is not in the document]
Comment (ram02z) [reply scripts/eval_context_quality.py:78]: To de-hyphenate by default, we need to set the flag when we get the page text.
So we need to update it to in `db.py`:
```python
page.get_text(flags=(TEXTFLAGS_TEXT | TEXT_DEHYPHENATE))
```

**Review by ram02z — COMMENTED** [review]
Looks good, I just think our method of parsing the PDF is too naive. Lets hold off for a while until we discuss this further.

**Comment by ram02z** [comment]
I've pushed an updated PDF parsing method using code from the [marker](ttps://github.com/VikParuchuri/) library. I've removed the OCR code, latex support, table support and post-processing to only keep the basic structure of the PDF excluding non-text items (like tables, pictures, equations). We then attempt to parse the section text and chunk each section to store as a document. The full parsing is fairly quick, doesn't take longer than 1 minute for short articles and 2 minutes for chapters.

I've tested the code on CPU on lecture 19 (the excel spreadsheet is on discord) and the results seem better on first glance.
If you are happy with this approach, I can add documentation about the process to the PR.

**Comment by wilsvd** [comment]
> @wilsvd I've pushed an updated PDF parsing method using code from the marker library. I've removed the OCR code, latex support, table support and post-processing to only keep the basic structure of the PDF excluding non-text items (like tables, pictures, equations). We then attempt to parse the section text and chunk each section to store as a document. The full parsing is fairly quick, doesn't take longer than 1 minute for short articles and 2 minutes for chapters.
> 
> I've tested the code on CPU on lecture 19 (the excel spreadsheet is on discord) and the results seem better on first glance. If you are happy with this approach, I can add documentation about the process to the PR.

Looking at the output, I'd say it looks alot nicer than it did before. In general seems alot better. I'll approve the pr.

File: scripts/eval_context_quality.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    # Save the workbook
+    workbook.save("source_vs_rag.xlsx")
+
+    # Delete database
+    db.client.reset()
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
```
Comment (ram02z) [inline scripts/eval_context_quality.py:128]: Ah, I forgot to remove this line. We should remove logging since we aren't using it anymore.

**Pull request merged**